src/libraries/System.Threading.Tasks.Dataflow/src/Blocks/TransformManyBlock.IAsyncEnumerable.cs

```diff
@@ -0,0 +1,303 @@
+            // through the reordering buffer when async processing completes,
+            // we know for certain that no more messages will need to be sent to the source.
+            _target.Completion.ContinueWith((completed, state) =>
+            {
+                var sourceCore = (SourceCore<TOutput>)state!;
+                if (completed.IsFaulted) sourceCore.AddAndUnwrapAggregateException(completed.Exception!);
+                sourceCore.Complete();
+            }, _source, CancellationToken.None, Common.GetContinuationOptions(), TaskScheduler.Default);
+
+            // It is possible that the source half may fault on its own, e.g. due to a task scheduler exception.
+            // In those cases we need to fault the target half to drop its buffered messages and to release its
+            // reservations. This should not create an infinite loop, because all our implementations are designed
+            // to handle multiple completion requests and to carry over only one.
+            _source.Completion.ContinueWith((completed, state) =>
+            {
+                var thisBlock = ((TransformManyBlock<TInput, TOutput>)state!) as IDataflowBlock;
+                Debug.Assert(completed.IsFaulted, "The source must be faulted in order to trigger a target completion.");
+                thisBlock.Fault(completed.Exception!);
+            }, this, CancellationToken.None, Common.GetContinuationOptions() | TaskContinuationOptions.OnlyOnFaulted, TaskScheduler.Default);
+
+            // Handle async cancellation requests by declining on the target
+            Common.WireCancellationToComplete(
+                dataflowBlockOptions.CancellationToken, Completion, state => ((TargetCore<TInput>)state!).Complete(exception: null, dropPendingMessages: true), _target);
+#if FEATURE_TRACING
+            DataflowEtwProvider etwLog = DataflowEtwProvider.Log;
+            if (etwLog.IsEnabled())
+            {
+                etwLog.DataflowBlockCreated(this, dataflowBlockOptions);
+            }
+#endif
+        }
+
+        /// <summary>Processes the message with a user-provided transform function that returns an async enumerable.</summary>
+        /// <param name="transformFunction">The transform function to use to process the message.</param>
+        /// <param name="messageWithId">The message to be processed.</param>
+        private void ProcessMessage(Func<TInput, IAsyncEnumerable<TOutput>> transformFunction, KeyValuePair<TInput, long> messageWithId)
+        {
+            Debug.Assert(transformFunction != null, "Function to invoke is required.");
+
+            bool userDelegateSucceeded = false;
+            try
+            {
+                // Run the user transform and store the results.
+                IAsyncEnumerable<TOutput> outputItems = transformFunction(messageWithId.Key);
+                userDelegateSucceeded = true;
+                ValueTask _ = StoreOutputItemsAsync(messageWithId, outputItems);
```

Comment:
This is not valid.  The task needs to be awaited.

Reply:
`ValueTask` is now awaited, but this leads to an `async void` method... dangerous, no? Is there a better "pattern" to do that?